main.nf

```diff
@@ -9,74 +9,110 @@
 ----------------------------------------------------------------------------------------
 */
 
-nextflow.enable.dsl = 2
-
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    GENOME PARAMETER VALUES
+    IMPORT FUNCTIONS / MODULES / SUBWORKFLOWS / WORKFLOWS
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-params.fasta = WorkflowMain.getGenomeAttribute(params, 'fasta')
-params.gtf = WorkflowMain.getGenomeAttribute(params, 'gtf')
-params.gff = WorkflowMain.getGenomeAttribute(params, 'gff')
-params.gene_bed = WorkflowMain.getGenomeAttribute(params, 'bed12')
-params.bwa_index = WorkflowMain.getGenomeAttribute(params, 'bwa')
-params.bwamem2_index = WorkflowMain.getGenomeAttribute(params, 'bwamem2')
-params.dragmap = WorkflowMain.getGenomeAttribute(params, 'dragmap')
-params.bowtie2_index = WorkflowMain.getGenomeAttribute(params, 'bowtie2')
+include { NASCENT } from './workflows/nascent'
+include { PIPELINE_INITIALISATION } from './subworkflows/local/utils_nfcore_nascent_pipeline'
+include { PIPELINE_COMPLETION     } from './subworkflows/local/utils_nfcore_nascent_pipeline'
+include { getGenomeAttribute      } from './subworkflows/local/utils_nfcore_nascent_pipeline'
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    VALIDATE & PRINT PARAMETER SUMMARY
+    GENOME PARAMETER VALUES
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-include { validateParameters; paramsHelp } from 'plugin/nf-validation'
-
-// Print help message if needed
-if (params.help) {
-    def logo = NfcoreTemplate.logo(workflow, params.monochrome_logs)
-    def citation = '\n' + WorkflowMain.citation(workflow) + '\n'
-    def String command = "nextflow run ${workflow.manifest.name} --input samplesheet.csv --genome GRCh37 -profile docker"
-    log.info logo + paramsHelp(command) + citation + NfcoreTemplate.dashedLine(params.monochrome_logs)
-    System.exit(0)
-}
-
-// Validate input parameters
-if (params.validate_params) {
-    validateParameters()
-}
-
-WorkflowMain.initialise(workflow, params, log, args)
+params.fasta = getGenomeAttribute('fasta')
+params.gtf = getGenomeAttribute('gtf')
+params.gff = getGenomeAttribute('gff')
+params.gene_bed = getGenomeAttribute('bed12')
+params.bwa_index = getGenomeAttribute('bwa')
+params.bwamem2_index = getGenomeAttribute('bwamem2')
+params.dragmap = getGenomeAttribute('dragmap')
+params.bowtie2_index = getGenomeAttribute('bowtie2')
+params.hisat2_index = getGenomeAttribute('hisat2')
+params.star_index = null
+params.homer_uniqmap = getGenomeAttribute('uniqmap')
```

Comment:
```suggestion
params.fasta                    = getGenomeAttribute('fasta')
params.gtf                        = getGenomeAttribute('gtf')
params.gff                        = getGenomeAttribute('gff')
params.gene_bed            = getGenomeAttribute('bed12')
params.bwa_index           = getGenomeAttribute('bwa')
params.bwamem2_index = getGenomeAttribute('bwamem2')
params.dragmap              = getGenomeAttribute('dragmap')
params.bowtie2_index     = getGenomeAttribute('bowtie2')
params.hisat2_index        = getGenomeAttribute('hisat2')
params.star_index            = null
params.homer_uniqmap   = getGenomeAttribute('uniqmap')
```

Reply:
Just going to wait for nextflow lint. I've held out this long 😬